PathManager/Src/PM_States.cpp

```diff
@@ -0,0 +1,374 @@
+ * Code
+ **********************************************************************************************************************/
+
+void CommsWithSystemManager::execute(pathManager* pathMgr)
+{
+    GetSMIncomingData(&incomingData);
+
+    if(pathMgr->isError)
+    {
+        pathMgr->setState(fatalFailureMode::getInstance());
+    }
+    else if (incomingData.waypoint_type == TELEOP_MODE)
+    {
+        pathMgr->flight_stage = TELEOP;
+        pathMgr->setState(CommsWithAttitude::getInstance()); 
+    }
+    else
+    {
+        pathMgr->setState(FlightModeSelector::getInstance());
+    }
+}
+
+pathManagerState& CommsWithSystemManager::getInstance()
+{
+    static CommsWithSystemManager singleton;
+    return singleton;
+}
+
+#ifdef UNIT_TESTING
+
+#include "cruisingMode.hpp"
+
+static Telemetry_Waypoint_Data_t createTelemetryWaypoint(long double lon, long double lat, int alt, float turnRadius, int type) {
+    Telemetry_Waypoint_Data_t waypoint {};
+
+    waypoint.longitude = lon;
+    waypoint.latitude = lat;
+    waypoint.altitude = alt;
+    waypoint.turnRadius = turnRadius;
+    waypoint.waypointType = type;
+
+    return waypoint;
+}
+
+#endif
+void FlightModeSelector::execute(pathManager* pathMgr){
+    if(pathMgr->isError)
+    {
+        pathMgr->setState(fatalFailureMode::getInstance());
+    }
+
+    // May need to go to a preflight stage later
+    /*if(CommsWithTelemetry::GetTelemetryIncomingData()->takeoffCommand && SensorFusion::GetSFOutput()->altitude < ON_GROUND_ALT)
+    {
+        pathMgr->flight_stage = PREFLIGHT;
+
+    }*/
+
+    // Prev code had a preflight stage for motor spin up...may be used later. 
+    //if (pathMgr->flight_stage == PREFLIGHT && PreflightStage::getPreFlightCompleteStatus){
+
+
+    // Checking if we are going from disarmed to armed to start takeoff stage. 
+    // TODO: Change and remove DISARMED state as SM will take care of this. 
+    if (CommsWithAttitude::GetCommFromAttitude()->armed && pathMgr->flight_stage == DISARMED){
+        pathMgr->flight_stage = TAKEOFF; 
+
+    }
+
+    // Check if altitude target has been reached for takeoff to start cruising stage. 
+    if(CommsWithSystemManager::GetSMIncomingData()->sf_data.altitude > LandingTakeoffManager::getTakeoffAltitudeTarget && pathMgr->flight_stage == TAKEOFF)
+    {
+        pathMgr->flight_stage = CRUISING;
+
+    }
+
+     if(CommsWithSystemManager::GetSMIncomingData()->landing_initiation.start_landing)
+    {
+        pathMgr->flight_stage = LANDING;
+    
+    }
+
+    // Assuming SF can give elapsed time, check if below landing target for a certain amount of time to start landed stage. 
+    if (CommsWithSystemManager::GetSMIncomingData()->sf_data.altitude < LandingTakeoffManager::getLandingAltitudeTarget && CommsWithSystemManager::GetSMIncomingData()->sf_data.deltaTime > LANDING_TIME_THRESHOLD){
+        pathMgr->flight_stage = LANDED; 
+    }
```

Comment:
This will likely exist in SM as SM can then shut off motors easier and shutting off motors doesn't require a waypoint.